perf/ring.go

```diff
@@ -158,3 +195,78 @@ func (rr *ringReader) Read(p []byte) (int, error) {
 
 	return n, nil
 }
+
+func (rr *ringReader) readOverwritable(p []byte) (int, error) {
+	read := rr.tail
+
+	// If size is 0, it means we read all the data avaiable in the buffer and jump
```

Comment:
> I nonetheless agree this is not perfect to have this here as we can call this function from other path than readRawSample().

That's what I'm trying to say, yes.

> I do not have this issue in Inspektor Gadget as I read the header and the sample in the same function:

You're assuming that the ring is zero filled, which is only true before anything is written to the ring. Later on you'll just have random garbage in there, since the kernel doesn't do any zeroing (afaik).

> I am not sure what would be a good solution here.

I think this case isn't needed since we can handle it like a partially overwritten record (which won't be zero), see below.